querier/lexer/lexer.go

```
@@ -0,0 +1,199 @@
+	}
+
+	l.readChar()
+	return tok
+}
+
+func (l *Lexer) readIdentifier() token.Token {
+	pos := l.pos
+
+	for {
+		// Stop if we hit a boundary: space, comma, EOF, or an operator (=, &, |, etc.)
+		if l.char == 0 || isWhitespace(l.char) || l.char == ',' || isOperator(l.char) {
```

golangci-lint (2.11.4)

[error] QF1006: could lift into loop condition

(staticcheck)